mpz-core/src/aes.rs

```diff
@@ -42,4 +42,93 @@ impl FixedKeyAes {
 
         Block::from(out)
     }
+
+    /// Correlation-robust hash function for 128-bit inputs
+    /// (cf. <https://eprint.iacr.org/2019/074>, §7.2).
+    /// The function computes `π(x) ⊕ x`.
+    /// π(x) = AES(fixedkey,x)
+    #[inline]
+    pub fn cr(&self, block: Block) -> Block {
+        let mut x = GenericArray::from(block);
+        self.aes.encrypt_block(&mut x);
+        Block::from(x) ^ block
+    }
+
+    /// Circular correlation-robust hash function
+    /// (cf.<https://eprint.iacr.org/2019/074>, §7.3).
+    ///
+    /// The function computes `H(sigma(x))`, where `H` is a correlation-robust hash
+    /// function and `sigma( x0 || x1 ) = (x0 xor x1) || x1`.
+    #[inline]
+    pub fn ccr(&self, block: Block) -> Block {
+        self.cr(Block::sigma(block))
+    }
+}
+
+/// A wrapper of aes, only for encryption.
+#[derive(Clone)]
+pub struct AesEncryptor(Aes128Enc);
+
+impl AesEncryptor {
+    /// AES_BLOCK_SIZE
+    pub const AES_BLOCK_SIZE: usize = 8;
```

Comment:
this var seems `prg.rs` specific. maybe it should live there?

Reply:
It will be used in other places, not just `Prg`, and it is related to `AES`.

Reply:
im ok to keep it here, can we maybe call it `AES_BLOCK_COUNT` and give it a more descriptive comment?

Reply:
@xiangxiecrypto ---^